src/main/java/org/gaul/s3proxy/EncryptedBlobStore.java

```diff
@@ -0,0 +1,784 @@
+            mbm.getContentMetadata().setContentLength(size);
+        } else {
+            // on s3 backends like aws or minio we rely on the eTag suffix
+            String regex = ".*-([0-9]+)";
+            Pattern pattern = Pattern.compile(regex);
+            Matcher matcher = pattern.matcher(blobMeta.getETag());
+            if (matcher.find()) {
+                int parts = Integer.parseInt(matcher.group(1));
+                int partPaddingSizes = Constants.PADDING_BLOCK_SIZE * parts;
+                long size = blobMeta.getSize() - partPaddingSizes;
+                mbm.setSize(size);
+                mbm.getContentMetadata().setContentLength(size);
+            } else {
+                long size = blobMeta.getSize() - Constants.PADDING_BLOCK_SIZE;
+                mbm.setSize(size);
+                mbm.getContentMetadata().setContentLength(size);
+            }
+        }
+
+        return mbm;
+    }
+
+    private boolean multipartRequiresStub() {
+        String blobStoreType = getBlobStoreType();
+        return Quirks.MULTIPART_REQUIRES_STUB.contains(blobStoreType);
+    }
+
+    private String blobNameWithSuffix(String container, String name) {
+        String nameWithSuffix = blobNameWithSuffix(name);
+        if (delegate().blobExists(container, nameWithSuffix)) {
+            name = nameWithSuffix;
+        }
+        return name;
+    }
+
+    private String blobNameWithSuffix(String name) {
+        return name + Constants.S3_ENC_SUFFIX;
+    }
+
+    private String getBlobStoreType() {
+        return delegate().getContext().unwrap().getProviderMetadata().getId();
+    }
+
+    private String generateUploadId(String container, String blobName) {
+        String path = container + "/" + blobName;
+        return DigestUtils.sha256Hex(path);
+    }
+
+    @Override
+    public Blob getBlob(String containerName, String blobName) {
+        try {
+            // adjust name if a blob with the .s3enc suffix exists
+            blobName = blobNameWithSuffix(containerName, blobName);
+
+            // init the decryption for a full blob
+            Decryption decryption =
+                new Decryption(secretKey, delegate(), containerName, blobName,
+                    0L, -1L);
+            return decryptBlob(decryption, containerName,
+                delegate().getBlob(containerName, blobName));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Blob getBlob(String containerName, String blobName,
+        GetOptions getOptions) {
+
+        try {
+            // adjust name if a blob with the .s3enc suffix exists
+            blobName = blobNameWithSuffix(containerName, blobName);
+
+            // init the decryption
+            Decryption decryption = null;
+            if (getOptions != null && getOptions.getRanges().size() > 0) {
+
+                // we support only one range
+                if (getOptions.getRanges().size() > 1) {
```

Comment:
This should never happen given the code here: https://github.com/gaul/s3proxy/blob/master/src/main/java/org/gaul/s3proxy/S3ProxyHandler.java#L1712-L1714. I don't mind being defensive, but since S3 doesn't allow multiple ranges, I'd expect the situation to never happen.